src/core/system.js

```diff
@@ -35,6 +35,7 @@ export default class Store {
     deepExtend(this, {
       state: {},
       plugins: [],
+      pluginsOptions: {},
```

Comment:
```suggestion
```

Reply:
along with previous suggested change, it should not be necessary to deepExtend the configuration object

Reply:
hmm, nevermind

Reply:
With further testing, I can confirm that we DO need to deepExtend the configuration object :)